src/opensynth/datasets/low_carbon_london/preprocess_lcl.py

```
@@ -130,56 +167,72 @@ def drop_dupes_and_replace_nulls(df: pd.DataFrame) -> pd.DataFrame:
+        by=["ID", "date", "settlement_period"], ascending=True
     )
     df_out = df_out.drop_duplicates(
-        subset=["LCLid", "date", "settlement_period"], keep="last"
+        subset=["ID", "date", "settlement_period"], keep="last"
     )
-    df_out["kwh"] = df_out["kwh"].replace("Null", np.float64())
-    df_out["kwh"] = df_out["kwh"].astype(float)
+    df_out = df_out.dropna()
     return df_out
 
 
-def filter_missing_kwh(df: pd.DataFrame) -> pd.DataFrame:
+def filter_missing_kwh(
+    df: pd.DataFrame, time_resolution: str = "half_hourly"
+) -> pd.DataFrame:
     """
-    Drop dates where we don't have full 48 readings
-    for a given LCLid and date.
+    Drop dates where we don't have full 48 readings if half-hourly data,
+    or 24 readings if hourly data, for a given ID and date.
 
     Args:
-        df (pd.DataFrame): _description_
+        df (pd.DataFrame): dataset
+        time_resolution (str): Time resolution of the data. Allowed values:
+            "half_hourly", "hourly".
 
     Returns:
-        pd.DataFrame: _description_
+        pd.DataFrame: filtered dataset
     """
     logger.info("🔍 Filtering missing kwh readings")
-    id_col = ["LCLid"]
-    merge_cols = id_col + ["date"]
+    merge_cols = ["ID", "date"]
     df_group = df.groupby(merge_cols)[["kwh"]].count().reset_index()
-    df_group["required_len"] = 48  # 48 hh readings
+
+    if time_resolution == "half_hourly":
+        required_len = 48  # 48 hh readings
+    elif time_resolution == "hourly":
+        required_len = 24  # 24 h readings
+    else:
+        raise ValueError("time_resolution must be 'half_hourly' or 'hourly'")
+    df_group["required_len"] = required_len
 
     df_full_data = df_group.query("required_len==kwh")  # Has all required data
     df_out = df_full_data[merge_cols].merge(df, on=merge_cols, how="inner")
+
+    assert len(df_out) > 0, "No data left after filtering missing kwh readings"
```

Instead of `assert` can you raise an error?

```python
if len(df_out) == 0:
    raise ValueError("No data left after filtering days with missing kWh readings")
```